e2e/README.md

```diff
@@ -145,7 +149,1089 @@ func E2ETests(env e2e.TestEnv) testhelper.Tests {
+      if the CPU architecture we're currently running on is among those listed
+      in `archs`
+  - See internal/pkg/test/tool/require/require.go for the full set of
+    require.XYZ() functions.
+
+- `user.CurrentUser(t *testing.T) *user.User`, defined in
+  e2e/internal/e2e/user.go, returns a struct with information about the current
+  user (UID, GID, home directory, etc.)
+  - The `user.User` struct is defined in internal/pkg/util/user/identity_unix.go
+
+- `tmpl.Execute(t *testing.T, tmpdir, namePattern, tmplPath string, values any)
+  string`, defined in internal/pkg/test/tool/tmpl/tmpl.go
+  - This is a convenience function for the use of [Go
+    templates](https://pkg.go.dev/text/template) in tests.
+    - Like the `require` package, it is available for unit-tests as well as e2e
+      tests.
+  - Execute() injects a given set of `values` into a template (whose path is
+    given by `tmplPath`), and places the result in a new, temporary file created
+    in `tmpdir` and named using the pattern `namePattern`.
+  - The created file is automatically removed at the end of the test `t`, unless
+    the test fails.
+
+## Common pitfalls
+
+**Scenario 1:** You've written your new test function, placed it in the right
+file (e.g. `imgbuild.go`), and now... your test doesn't seem to be running. You
+can find it in the e2e output logs anywhere.
+
+**Common cause:** You've forgotten to add your test function to the
+`testhelper.Tests` map created by your testing group's E2ETests() function.
+
+This function is typically located at the bottom of the Go source file
+corresponding to the e2e group (see the [introduction](#introduction) for an
+example). Your test function - really, a method of the group's e2e context
+object - won't run unless it is added to the `testhelper.Tests` map.
+
+Make sure to also [mark your test
+appropriately](#parallel-par-vs-non-parallel-seq-tests) if it cannot be run in
+parallel.
+
+**Scenario 2:** You're trying to get the e2e test to only run your test. You've
+set the [E2E_GROUPS and E2E_TESTS](#running-the-e2e-suite) environment
+variables, and you've double checked the spelling of the group & test names, but
+it's not catching your test; nothing is running.
+
+**Common cause:** You're trying to use E2E_TESTS to filter tests by something
+*other* than the top-level test name. Here is an example of a test name from a
+specific e2e subtest:
+
+```plain
+TestE2E/SEQ/INSTANCE/auth/Root/start_before_auth
+```
+
+`TestE2E` is the single top-level name for all e2e tests; next comes [`PAR` or
+`SEQ`](#parallel-par-vs-non-parallel-seq-tests), then comes the group name
+(which is what E2E_GROUPS matches), followed by the top-level test name (which
+is what E2E_TESTS matches; in this case, `auth`). The E2E_TESTS environment
+variable is a regular expression that is matched only against the top-level test
+name.
+
+At the present, there is no way to match against more deeply-embedded components
```

Comment:
You can actually do this with `make -C builddir e2e-test -run /my/specific/test/name/here`